### DISCUSSION TITLE
Simplification of the AEIP9

### DISCUSSION THREAD

**Discussion by samuelmanzanera** [archethic-foundation/aeip#46] [Ideas]
Since the AEIP9 we introduced a structure to reference assets in token with base64, AEWeb,  IPFS or HTTP links.
However this structure is a bit cumbersome and could be lifted a bit.
By using specific URI scheme we could reduce the footprint of the token def reducing fees reduce its cumbersomeness

What do you think of URI scheme like those:
- aeweb://0000f2c75f55fb398e99b6a5e674c3422731fb167f79b9b08131dcfa3bb830d50b9a
- https://my-website.com/image.png
- ipfs://bafybeihkoviema7g3gxyt6la7vd5ho32ictqbilu3wnlo3rs7ewhnp7lly
- data:image/gif;base64,R0lGODlhEAAQAMQAAORHHOVSKudfOulrSOp3WOyDZu6QdvCchPGolfO0o/XBs/fNwfjZ0frl3/zy7////wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACH5BAkAABAALAAAAAAQABAAAAVVICSOZGlCQAosJ6mu7fiyZeKqNKToQGDsM8hBADgUXoGAiqhSvp5QAnQKGIgUhwFUYLCVDFCrKUE1lBavAViFIDlTImbKC5Gm2hB0SlBCBMQiB0UjIQA7

The latter follow the Data URI scheme which well known in the web industry and could support different kind of format, hence removing the need of the type_mime field for this embedded resource.

**Comment by Neylix** [comment]
Could be a good idea, but the type_mime would still be needed for aeweb / https / ipfs link. 
Could we directly integrate the mime in the url as for data :
- aeweb:image/png;0000f2c75f55fb398e99b6a5e674c3422731fb167f79b9b08131dcfa3bb830d50b9a
- https:image/png;my-website.com/image.png
- ipfs:image/png;bafybeihkoviema7g3gxyt6la7vd5ho32ictqbilu3wnlo3rs7ewhnp7lly
- data:image/gif;base64,R0lGODlhEAAQAMQAAORHHOVSKudfOulrSOp3WOyDZu6QdvCchPGolfO0o/XBs/fNwfjZ0frl3/zy7////wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACH5BAkAABAALAAAAAAQABAAAAVVICSOZGlCQAosJ6mu7fiyZeKqNKToQGDsM8hBADgUXoGAiqhSvp5QAnQKGIgUhwFUYLCVDFCrKUE1lBavAViFIDlTImbKC5Gm2hB0SlBCBMQiB0UjIQA7

As it the type_mime field could be completely removed

**Comment by bchamagne** [comment]
I like this simplification. 

> but the type_mime would still be needed for aeweb / https / ipfs link.

I don't get why you'd need mime for aeweb/https/ipfs. Only the data require it.

**Reply by Neylix** [reply]
As an application developer when I fetch the content of a NFT I would like to know how to display it, I would display the content in different manner if it's a image or pdf or video or whetever else.
I would also filter the type to avoid downloading something I cannot display

**Reply by bchamagne** [reply]
Got it, cheers

**Comment by samuelmanzanera** [comment]
I'm not against to keep it, if we want to keep the mime_type, we can:
- do a custom scheme: `base64://.....`
- skip the mime type in the dataURI: data:;base64,R0lGODlhEAAQAMQAAORHHOVSKudfOulrSOp3WOyDZu6QdvCchPGolfO0o/XBs/fNwfjZ0frl3/zy7////wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACH5BAkAABAALAAAAAAQABAAAAVVICSOZGlCQAosJ6mu7fiyZeKqNKToQGDsM8hBADgUXoGAiqhSvp5QAnQKGIgUhwFUYLCVDFCrKUE1lBavAViFIDlTImbKC5Gm2hB0SlBCBMQiB0UjIQA7

WIth this latest option, the application wishing the be dataURI compliant to render the image, it would have to inject the mime-type before the ';'
